app/src/main/java/io/intrepid/skeleton/base/BasePresenter.java

```diff
@@ -0,0 +1,71 @@
+package io.intrepid.skeleton.base;
+
+import android.support.annotation.NonNull;
+
+import io.intrepid.skeleton.rest.RestApi;
+import io.intrepid.skeleton.settings.UserSettings;
+import rx.Observable;
+import rx.Scheduler;
+import rx.subscriptions.CompositeSubscription;
+
+public abstract class BasePresenter<T extends BaseContract.View> implements BaseContract.Presenter<T> {
+
+    protected final CompositeSubscription subscriptions = new CompositeSubscription();
+
+    protected T view;
+    @NonNull
+    protected final Scheduler ioScheduler;
+    @NonNull
+    protected final Scheduler uiScheduler;
+    @NonNull
+    protected final UserSettings userSettings;
+    @NonNull
+    protected final RestApi restApi;
+
+    public BasePresenter(@NonNull T view, @NonNull PresenterConfiguration configuration) {
+        this.view = view;
+        this.ioScheduler = configuration.getIoScheduler();
+        this.uiScheduler = configuration.getUiScheduler();
+        this.userSettings = configuration.getUserSettings();
+        this.restApi = configuration.getRestApi();
+    }
+
+    @Override
+    public void onViewCreated() {
+
+    }
+
+    @Override
+    public final void bindView(T view) {
+        this.view = view;
+        onViewBinded();
+    }
```

Comment:
Add a lock to `bindView(T view)` and `unbindView()`. This will force `onViewBound()` and `onViewUnbound()`  to hopefully only be called once. We call `bindView(T view)` in `onResume()` which is called multiple times when committing fragment transactions. (https://github.com/IntrepidPursuits/MITMobius-android/blob/develop/app/src/main/java/edu/mit/mobius/app/presenter/BaseMVP.java)